holopy/inference/tests/test_prior.py

```diff
@@ -17,60 +17,320 @@
-    assert_equal(u.guess, 1.5)
-    assert_equal(u.interval, 1)
-    assert_raises(ParameterSpecificationError, Uniform, 4, 6, 7)
-    assert_raises(ParameterSpecificationError, Uniform, 4, 4)
-
-def test_improper_guess():
-    improper1 = Uniform(-2, np.inf)
-    assert_equal(improper1.guess, -2)
-    improper2 = Uniform(-2, np.inf, -1)
-    assert_equal(improper2.guess, -1)
-    improper3 = Uniform(-np.inf, -2)
-    assert_equal(improper3.guess,-2)
-    improper4 = Uniform(-np.inf, np.inf)
-    assert_equal(improper4.guess, 0)
-
-def test_bounded_gaussian():
-    g = prior.BoundedGaussian(1, 1, 0, 2)
-    assert_equal(g.lnprob(-1), -np.inf)
-    assert_equal(g.lnprob(3), -np.inf)
-    assert_equal(g.guess, 1)
-
-def test_gaussian():
-    g = Gaussian(1, 1)
-    assert_equal(g.guess, 1)
-    assert_obj_close(g.lnprob(0),gold_sigma)
-
-def test_complex_prior():
-    p1 = prior.ComplexPrior(Uniform(0,2), Gaussian(2,1))
-    assert_obj_close(p1.real, Uniform(0,2))
-    assert_obj_close(p1.imag, Gaussian(2,1))
-    assert_equal(p1.guess, 1+2j)
-    assert_obj_close(p1.lnprob(0.5+1j), gold_sigma - np.log(2))
-    p2 = prior.ComplexPrior(1, Gaussian(2,1))
-    assert_equal(p2.guess, 1+2j)
-    p3 = prior.ComplexPrior(Uniform(0,2), 2)
-    assert_equal(p3.guess, 1+2j)
+
+GOLD_SIGMA = -1.4189385332 #log(sqrt(0.5/pi))-1/2
+
+class TestBasics(unittest.TestCase):
+    @attr("fast")
+    def test_cannot_instantiate_baseclass(self):
+        self.assertRaises(NotImplementedError, Prior)
+
+class TestUniform(unittest.TestCase):
+    @attr("fast")
+    def test_construction(self):
+        parameters = {'lower_bound':1, 'upper_bound':3, 'guess':2, 'name':'a'}
+        u = Uniform(*parameters.values())
+        self.assertTrue(isinstance(u, Prior))
+        for key, val in parameters.items():
+            self.assertEqual(getattr(u, key), val)
+
+    @attr("fast")
+    def test_upper_bound_larger_than_lower_bound(self):
+        self.assertRaises(ParameterSpecificationError, Uniform, 1, 0)
+        self.assertRaises(ParameterSpecificationError, Uniform, 1, 1)
+
+    @attr("fast")
+    def test_guess_must_be_in_interval(self):
+        self.assertRaises(ParameterSpecificationError, Uniform, 0, 1, 2)
+        self.assertRaises(ParameterSpecificationError, Uniform, 0, 1, -1)
+
+    @attr("fast")
+    def test_interval_calculation(self):
+        bounds = np.random.rand(2) + np.array([0,1])
+        u = Uniform(*bounds)
+        self.assertEqual(u.interval, np.diff(bounds))
+
+    @attr("fast")
+    def test_interval_is_property(self):
+        bounds = np.random.rand(2) + np.array([0,1])
+        u = Uniform(*bounds)
+        self.assertRaises(AttributeError, setattr, u, 'interval', 2)
+
+    @attr("fast")
+    def test_prob(self):
+        bounds = np.random.rand(2) + np.array([0,1])
+        u = Uniform(*bounds)
+        self.assertEqual(u.prob(0), 0)
+        self.assertEqual(u.prob(2), 0)
+        self.assertEqual(u.prob(1), 1/np.diff(bounds))
```

Comment:
Hmmm... I'm surprised this line doesn't have roundoff errors. I imagine that it's because you're calling the exact same operation in setting up the ```Uniform``` prior. But it might be worthwhile to change this to an
```python
self.assertAlmostEqual(u.prop(1), 1 / np.diff(bounds), places=14)
```
or something, in case someone changes ```Uniform``` to use some mathematically equivalent but numerically different slick new method.

Again, I'd rather have many good tests than a few perfect ones at this point, so up to you.